src/containers/task-management/task_detail.tsx

```diff
@@ -0,0 +1,446 @@
+                    </DescriptionListGroup>
+                  </DescriptionList>
+                </section>
+              </FlexItem>
+              <FlexItem>
+                <section className='body card-area'>
+                  <Title headingLevel='h2' size='lg'>
+                    {_`Reserve resources`}
+                  </Title>
+                  <br />
+                  {!!resources.length ? (
+                    <DescriptionList isHorizontal>
+                      {resources.map((resource, index) => {
+                        return (
+                          <React.Fragment key={resource.type + index}>
+                            <hr />
+                            <DescriptionListGroup>
+                              <DescriptionListTerm>{_`Type`}</DescriptionListTerm>
+                              <DescriptionListDescription>
+                                {resource.type}
+                              </DescriptionListDescription>
+                            </DescriptionListGroup>
+                            {resource.name && (
+                              <DescriptionListGroup>
+                                <DescriptionListTerm>{_`Name`}</DescriptionListTerm>
+                                <DescriptionListDescription>
+                                  {resource.name}
+                                </DescriptionListDescription>
+                              </DescriptionListGroup>
+                            )}
+                          </React.Fragment>
+                        );
+                      })}
+                    </DescriptionList>
+                  ) : (
+                    _`There's no resource record`
+                  )}
+                </section>
+              </FlexItem>
+            </Flex>
+            <Flex
+              direction={{ default: 'column' }}
+              flex={{ default: 'flex_1' }}
+            >
+              <FlexItem>
+                {!task.error && (
+                  <section className='body card-area'>
+                    <Title headingLevel='h2' size='lg'>
+                      {_`Progress messages`}
+                    </Title>
+                    <br />
+                    {!!task.progress_reports.length ? (
+                      <DescriptionList isHorizontal>
+                        {task.progress_reports
+                          .reverse()
+                          .map((report, index) => {
+                            return (
+                              <React.Fragment key={index}>
+                                <hr />
+                                {Object.keys(report).map((key, index) => {
+                                  return (
+                                    !!report[key] && (
+                                      <DescriptionListGroup key={key + index}>
+                                        <DescriptionListTerm>
+                                          {capitalize(key)}
```

Comment:
Sounds like maybe we know the keys now?
What about something like..

```suggestion
                                          {{ message: t`Message`, code: t`Code`, state: t`State`, done: t`Done` }[key] || capitalize(key)}
```

(+- linter :))